fix(swap): apply swap delay in swap function instead of handleAjaxResponse

boonedox opened the pull request:
## Description

moved application of swap delay to swap function so that the JS api can also use the swap delay

Corresponding issue:

## Testing
manually tested in my local environment, both with the JS api and with normal hx requests

## Checklist

* [ ] I have read the contribution guidelines
* [ ] I have targeted this PR against the correct branch (`master` for website changes, `dev` for
  source changes)
* [ ] This is either a bugfix, a documentation update, or a new feature that has been explicitly
  approved via an issue
* [ ] I ran the test suite locally (`npm run test`) and verified that it succeeded

Telroshan commented:
Hey,
- You'll want to rebase against `dev` (not merge) to drop unnecessary commits that currently pollute your PR's diff
- Could you add test cases for this?
Thanks!

boonedox commented:
> Hey,
> 
> * You'll want to rebase against `dev` (not merge) to drop unnecessary commits that currently pollute your PR's diff
> * Could you add test cases for this?
>   Thanks!

I've rebased and added a test that fails on `dev` and works on the bugfix branch

Telroshan commented:
Thanks!
Took the liberty to add a link to [this issue](https://github.com/bigskysoftware/htmx-extensions/issues/127) (opened in the extensions repo) to your PR description for other reviewers btw, as it's directly concerned by this suggestion!

RobMaskell commented:
Any news on this getting merged as this fixes a problem I was having?

1cg commented:
i can accept this
